packages/smooth_app/lib/pages/product/common/product_list_item_simple.dart

```diff
@@ -68,7 +70,8 @@ class _ProductListItemSimpleState extends State<ProductListItemSimple> {
             case LoadingStatus.LOADED:
               if (_model.product != null) {
                 return SmoothProductCardFound(
-                  heroTag: _model.product!.barcode!,
+                  heroTag:
+                      '${_model.product!.barcode!}_${Random().nextInt(100)}',
```

Comment:
This will lead to no hero animation at all, why do we need it to be random here

Reply:
I totally agree with you @M123-dev: the only added value of `Hero` is when you use common keys betweens pages. If we use only different keys, that's just some extra useless code to maintain.
Btw I can't remember even seeing a single `Hero` animation on Smoothie.
My suggestion (again): just get rid of `Hero`.

Reply:
In the latest version there seems to be a very well visible one (opening a product from a list) and I quite like it, also a one line animation, that's as simple as it gets.

Reply:
I think you've misunderstood how the app behaves now AND with my changes.

Today, when we open the ProductPage, on the current page, we may have a Hero widget, where the tag is the barcode, ensuring the unicity. And on the ProductPage, we have another Hero, where the tag is also the same barcode. And basically, Flutter will match the two tags.

However this leads to an issue, where in the Widgets' tree, we can have 2 or 3 Hero from different search views with the same barcode (eg: a search view and the history). That's why we have to differentiate them, hence the use of a Random and a prefix.

And the other change in my code, is to pass the custom tag to the Product page, to ensure both sides will use the correct ID.

Reply:
> I think you've misunderstood how the app behaves now AND with my changes.

That's possible.

> And the other change in my code, is to pass the custom tag to the Product page, to ensure both sides will use the correct ID.

That was the part I missed. That makes more sense.

Still, with my limited appetite for animations, I would prefer a code without the tap dancing about `Hero` tags. Just no `Hero` to maintain. But that's a matter of personal coder taste.